### PR TITLE
Fix function name parsing with whitespace

### PR DESCRIPTION
**Pull request by ntkme**
This PR fixes function name parsing when whitespace exists between function name and `(`.

The following code works on sass but not sass-embedded due to function name being incorrectly parsed as `x ` (with extra whitespace at the end):

```
const sass = require('sass-embedded')

const result = sass.compileString('a{b:x()}', {
  functions: {
    'x ()': function () { return sass.sassTrue }
  }
})

console.log(result)
```

https://github.com/sass/sass-spec/pull/1854

### CODE REVIEW

**Comment by ntkme** [comment]
@nex3 Any comments?

**Comment by nex3** [comment]
Sorry for the delay, we've got a lot of irons in the fire at the moment.

This seems like something that should be an error, since this isn't a valid function signature. [The spec](https://github.com/sass/sass/blob/2c655fcf43184746d952960de4921a1aa30771e9/spec/js-api/options.d.ts#L59-L86) is unfortunately not very clear about this, so we should update that. Unfortunately, since Dart Sass supports this we'll need to do a deprecation process.

The upshot here is that this should probably produce a deprecation warning, but we can do that in a separate PR once the language update goes through.

File: lib/src/function-registry.ts [hunk 1]
```diff
@@ -57,11 +57,11 @@ export class FunctionRegistry<sync extends 'sync' | 'async'> {
   call(
     request: OutboundMessage.FunctionCallRequest
   ): PromiseOr<InboundMessage.FunctionCallResponse, sync> {
-    const protofier = new Protofier(this);
-    const fn = this.get(request);
-
     return catchOr(
       () => {
+        const protofier = new Protofier(this);
+        const fn = this.get(request);
```
Comment (nex3) [inline lib/src/function-registry.ts:63]: Why this change?

Comment (ntkme) [reply lib/src/function-registry.ts:63]: Because the error of this.get is not catches and raised back to compiler, the compilation will get stuck.

**Comment by ntkme** [comment]
Most programming language allows a whitespace between function name and `(` so I think this is actually acceptable.

**Comment by ntkme** [comment]
Sounds good. I guess since we are going to do deprecation, we should still want to make it work in sass-embedded like dart-sass for now, in a form that the complication does not get stuck, and then have both implementations behave the same on deprecation warnings.

**Comment by nex3** [comment]
I'm just going to make it a breaking change—since the spec already says it should behave this way, I don't feel too bad about it.
